src/cmdlib.sh

```diff
@@ -100,8 +100,11 @@ prepare_build() {
     preflight
     if ! [ -d repo ]; then
         fatal "No $(pwd)/repo found; did you run coreos-assembler init?"
-    elif ! has_privileges && [ ! -f cache/cache.qcow2 ]; then
-        fatal "No cache.qcow2 found; did you run coreos-assembler init?"
+    elif ! has_privileges; then
+        if [ ! -f cache/cache.qcow2 ]; then
+            qemu-img create -f qcow2 cache/cache.qcow2 10G
+            LIBGUESTFS_BACKEND=direct virt-format --filesystem=xfs -a cache/cache.qcow2
+        fi
```

Comment:
almost wonder if this wouldn't be better served in `cmd-build` file. this `prepare_build` function gets called in a lot of places.

Reply:
> almost wonder if this wouldn't be better served in `cmd-build` file. this `prepare_build` function gets called in a lot of places.

IMHO, I would vote for that. COSA does a lot and the cache is specific to the build function.

Reply:
Good point.  I'll do another patch.  (Though to test it I need to try out the `COREOS_ASSEMBLER_GIT` path for the first time as my usual dev workflow is privileged)

Reply:
Sorry, I missed this convo before merging.

Not against, though note the cache is used in both `fetch` and `build`. It outlives both invocations. To me, that points a little more towards doing it upfront.